go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,928 @@
+	eips, err := c.eIPLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list egress IPs: %v", err)
+	}
+	return eips, nil
+}
+
+//TODO: convert the following funcs to a startup and periodic cleanfunc to reduce risk of stale config
+//func (c *Controller) cleanNode() {
+//	if linkAddresses, linkRoutes, err := c.getLinkAddressRoutes(); err != nil {
+//		klog.Errorf("Failed to reconcile and ensure no stale egress IP addresses and routes: %v", err)
+//	} else {
+//		c.removeStaleAddresses(linkAddresses, nil)
+//		c.removeStaleRoutes(linkRoutes, nil)
+//	}
+//	if routingRules, err := getRoutingRules(); err != nil {
+//		klog.Errorf("Failed to reconcile and ensure no stale routing rules: %v", err)
+//	} else {
+//		c.removeStaleRoutingRules(routingRules, nil)
+//	}
+//	if c.v4 {
+//		if err := c.iptablesManager.FlushChain(utiliptables.TableNAT, iptChainName, utiliptables.ProtocolIPv4); err != nil {
+//			klog.Errorf("Failed to flush IPv4 rules in chain %q: %v", iptChainName, err)
+//		}
+//	}
+//	if c.v6 {
+//		if err := c.iptablesManager.FlushChain(utiliptables.TableNAT, iptChainName, utiliptables.ProtocolIPv6); err != nil {
+//			klog.Errorf("Failed to flush IPv6 rules in chain %q: %v", iptChainName, err)
+//		}
+//	}
+//}
+
+//
+//func (c *Controller) removeStaleAddresses(existingAddresses sets.Set[string]) error {
+//	foundAddresses := sets.New[string]()
+//	links, err := netlink.LinkList()
+//	if err != nil {
+//		return err
+//	}
+//	for _, link := range links {
+//		addresses, err := netlink.AddrList(link, netlink.FAMILY_ALL)
+//		if err != nil {
+//			return err
+//		}
+//
+//	}
+//	// remove any invalid addresses or stale egress IPs
+//	var isValidEIP bool
+//	for linkName, addresses := range foundLinkAddresses {
+//		for _, linkAddress := range addresses {
+//			if linkAddress.Label != linkmanager.GetAssignedAddressLabel(linkName) {
+//				continue
+//			}
+//			isValidEIP = false
+//			for _, validAddress := range validAddresses {
+//				if validAddress.IP.Equal(linkAddress.IP) {
+//					isValidEIP = true
+//					break
+//				}
+//			}
+//
+//			if !isValidEIP {
+//				if err := c.linkManager.DelAddress(linkAddress); err != nil {
+//					klog.Errorf("Failed to delete egress IP %s to link manager: %v", linkAddress.IP.String(), err)
+//				}
+//			}
+//		}
+//	}
+//}
+
+func getRouteTableID(ifIndex int) int {
+	return ifIndex + routingTableIDStart
+}
+
+// GetLinkContainingIP iterates through all links found locally building a map of addresses associated with
+// each link and attempts to find a network that will host the func parameter IP address using longest-prefix-match.
+// Expected network is validated against calculated network to ensure they match, otherwise, we return an error.
+// If no network is found to host the func parameter IP, no error is returned.
+func (c *Controller) GetLinkContainingIP(ip net.IP) (bool, netlink.Link, error) {
```

Comment:
name is misleading. Suggest `FindLinkOnSameSubnetAsIP` or something like that.

Reply:
Done